### PR TITLE
Sherry Xie Personal Page

### PR DESCRIPTION
**Pull request by sherryycxie**
# *Sherry's Views*

## :recycle: Current situation & Problem
I added in my own personal view to this submission.

## :bulb: Proposed solution
There shouldn't be any problem to the current situation.

## :gear: Release Notes 
In my personal view, I created features which allows users to click on a button to access different pages or slide between different pages. I consulted this YouTube video: https://www.youtube.com/watch?v=OPTzgeIKMgU for the functionality.

I also included images for each of my pages. All images are taken by me except for the image of Shanghai, referenced here: https://www.china-briefing.com/news/investing-in-shanghai-china-city-spotlight/.

## :heavy_plus_sign: Additional Information
N/A

### Related PRs
I referenced the image bundle within the Instructor Views package provided by Paul.

### Testing
Yes, I included testing. I tested all the buttons I have on my personal pages to make sure they work.

### Reviewer Nudging
It would be great by starting from the Member Views page and clicking on my name. @vishnuravi my page is ready for review!

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/CS342/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/CS342/.github/blob/main/CONTRIBUTING.md):
- [Yes] I agree to follow the [Code of Conduct](https://github.com/CS342/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/CS342/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: CS342MemberViews/Sources/SherryXie/SherryDescriptionView.swift
Comment (PSchmiedmayer) [inline CS342MemberViews/Sources/SherryXie/SherryDescriptionView.swift:24]: Really great that you have added documentation to your view! 🚀

File: CS342MemberViews/Sources/SherryXie/SherryPageModel.swift
Comment (PSchmiedmayer) [inline CS342MemberViews/Sources/SherryXie/SherryPageModel.swift:25]: As all of these properties are never changed, I would suggest making them all constants:
```suggestion
    static let samplePage = Page(name: "Title Example", description: "This is a sample description", imageUrl: "work", tag: 0)
    
    static let samplePages: [Page] = [
        Page(name: "Sherry Xie", description: "SherryBio", imageUrl: "SherryXie", tag: 0),
        Page(name: "Shanghai, China", description: "ShanghaiBio", imageUrl: "Shanghai", tag: 1),
        Page(name: "Culver, Indiana", description: "CulverBio", imageUrl: "Culver", tag: 2),
        Page(name: "Paris, France", description: "ParisBio", imageUrl: "Paris", tag: 3)
    ]
    
    let id = UUID()
    let name: String
    let description: String
    let imageUrl: String
    let tag: Int
```

File: CS342MemberViews/Sources/SherryXie/SherryPageModel.swift
Comment (PSchmiedmayer) [inline CS342MemberViews/Sources/SherryXie/SherryPageModel.swift:11]: Really great job structuring your models independent of you views, a great job! 👍

File: CS342MemberViews/Sources/SherryXie/SherryView.swift
Comment (PSchmiedmayer) [inline CS342MemberViews/Sources/SherryXie/SherryView.swift:12]: Great job here!

File: CS342MemberViews/Sources/SherryXie/SherryXie.swift
Comment (PSchmiedmayer) [inline CS342MemberViews/Sources/SherryXie/SherryXie.swift:36]: Great job with your view. I have noted that the buttons and the page view indicators overlap in the application. 
You can address this using a `padding` to your button. You an also remove the duplicated modifiers by packaging all the Buttons in a `Group`:
```suggestion
                    Group {
                        if page == pages.last {
                            Button("Go back to my home page", action: goToZero)
                        } else if page.tag == 0 {
                            Button("Check out the places I love!", action: incrementPage)
                        } else {
                            Button("Next", action: incrementPage)
                        }
                    }
                        .buttonStyle(.bordered)
                        .padding(.bottom, 50)
```

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for your great PR! I like the view and it looks great! Amazing job structuring your code in models and making sure that it is easy to read!

I have added few small comments that you can address.
I have also merged the latest main state in your branch to make merging the PR easy as you have already resolved one merge conflict already 👍

Be sure that you `git pull` the latest state to your local version if you accept any of the suggestions here on GitHub and get the merge conflict to your local branch 👍

**Comment by sherryycxie** [comment]
@PSchmiedmayer Hi Paul! I took in your suggestions and made those edits on my local branch. After merging the main branch with my branch, I found out that I started failing some of the tests because some of Audrey's files did not have license agreements. What do I do here?

**Comment by sherryycxie** [comment]
> @PSchmiedmayer Hi Paul! I took in your suggestions and made those edits on my local branch. After merging the main branch with my branch, I found out that I started failing some of the tests because some of Audrey's files did not have license agreements. What do I do here?

Actually, I just saw that Vishnu is working on the repository, fixing these conflicts. Will see how that goes!

**Comment by vishnuravi** [comment]
Hi Sherry, your branch is updated. Should pass REUSE now.

**Comment by sherryycxie** [comment]
> Hi Sherry, your branch is updated. Should pass REUSE now.

Sounds good! Paul already approved my changes, so can I just enable auto-merge once all tests pass?

**Comment by vishnuravi** [comment]
> > Hi Sherry, your branch is updated. Should pass REUSE now.
> 
> Sounds good! Paul already approved my changes, so can I just enable auto-merge once all tests pass?

Go for it! Nice job!

**Comment by PSchmiedmayer** [comment]
Great job @sherryycxie and thank you for your support @vishnuravi.
I have updated the branch to the latest version and will merge it when all builds pass 🚀

**Pull request merged**
